src/Compilers/CSharp/Portable/Symbols/TypeParameterBounds.cs

```diff
@@ -17,6 +17,16 @@ internal sealed class TypeParameterBounds
     {
         public static readonly TypeParameterBounds Unset = new TypeParameterBounds();
 
+        /// <summary>
+        /// Creates a "early" bound instance that has constraint types set
+        /// but no other fields.
+        /// </summary>
+        public TypeParameterBounds(ImmutableArray<TypeWithAnnotations> constraintTypes)
```

Comment:
>TypeParameterBounds [](start = 15, length = 19)

Would it make sense to keep this constructor private and to add a factory method with very specific name for consumers to use?